packages/lodestar/src/tasks/tasks/archiveStates.ts

```diff
@@ -38,10 +38,10 @@ export class ArchiveStatesTask implements ITask {
     this.logger.info(
       `Started archiving states (finalized epoch #${this.finalizedCheckpoint.epoch})...`
     );
-    // store the last available state in the finalized epoch
+    // store the state of finalized checkpoint
     const finalizedState = (await this.db.stateCache.values())
       .filter((state) => computeEpochAtSlot(this.config, state.slot) === this.finalizedCheckpoint.epoch)
-      .sort((a, b) => b.slot - a.slot)[0];
+      .sort((a, b) => a.slot - b.slot)[0];
```

Comment:
I want to store state of finalized check point so when I restore I'm confident with the finalized block based on `block.message.stateRoot = hashTreeRoot(lastKnownState)`.

Reply:
Don't you actually need first state where computeEpochAtSlot(this.config, state.slot) === this.finalizedCheckpoint.epoch) + 1?

Reply:
I need the 1st state to be the state of the finalized check point so that I initialize the state cache with that. When we reprocess block, we still need `preState` to exist. So it's like:
+ Store state of finalized checkpoint here
+ When chain is started, find finalized block based on `block.message.stateRoot = hashTreeRoot(lastKnownState)`
+ Init forkchoice, init state cache with finalized block, finalized state
+ Start processing blocks following the finalized block

Reply:
I guess the confusing thing is that only the first block in a finalized epoch is finalized. The rest are are not finalized even tho the epoch is finalized. This is bc  the epoch processing happens at the beginning of the epoch, rather than at the end. So any further blocks processed happen after the finalization has already occurred.
this lgtm tho